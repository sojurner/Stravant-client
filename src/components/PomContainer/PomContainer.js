import React, { Component } from 'react';
import { connect } from 'react-redux';
import { removePom } from '../../actions/pomAction';

import './PomContainer.css';

export class PomContainer extends Component {
  render() {
    const { pomHistory, removePom } = this.props;
    const pomContainer = Object.keys(pomHistory).map(time => {
      const pomTime = ` ${pomHistory[time].hour}h ${pomHistory[time].minute}m ${
        pomHistory[time].second
      }s`;
      return (
        <div className="pom-cards">
          <i class="fab fa-xbox" onClick={() => removePom(pomHistory, time)} />
          <h4>{time} </h4>
          <p>{pomTime}</p>
        </div>
      );
    });
    return <div className="pom-container">{pomContainer}</div>;
  }
}

const mapStateToProps = state => ({
  pomHistory: state.pomInfo.pomHistory
});

const mapDispatchToProps = dispatch => ({
  removePom: (history, time) => dispatch(removePom(history, time))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PomContainer);
