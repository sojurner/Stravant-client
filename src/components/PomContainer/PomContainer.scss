.pom-container {
  display: flex;
  flex-direction: horizontal;
  overflow-x: scroll;
  overflow-y: hidden;
  width: 20rem;
  font-family: daybold;
}

.pom-cards {
  min-width: 10rem;
  max-width: 10rem;
  padding: 0.1rem 0.5rem;
  margin: 0 0.2rem;
  border: 2px solid black;
  background-color: #acaaaa;
  font-size: 1.5rem;
}

.pom-cards:hover {
  background-color: #d0c4c4;
}

.fa-xbox {
  color: #7c564d;
}

.fa-xbox:hover {
  color: #c27062;

  cursor: pointer;
}
