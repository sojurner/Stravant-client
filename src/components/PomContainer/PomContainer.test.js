import React from 'react';
import { PomContainer } from './PomContainer';

describe('PomContainer', () => {
  let wrapper;
  let mockPomHistory;
  let mockRemove;

  beforeEach(() => {
    mockPomHistory = [
      {
        id: 473825,
        resource_state: 2,
        name: 'Stravant Club',
        profile_medium:
          'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246543/1/medium.jpg',
        profile:
          'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246543/1/large.jpg',
        cover_photo:
          'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246574/1/large.jpg',
        cover_photo_small:
          'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246574/1/small.jpg',
        sport_type: 'running',
        city: 'Denver',
        state: 'Colorado',
        country: 'United States',
        private: false,
        member_count: 4,
        featured: false,
        verified: false,
        url: 'stravant'
      }
    ];

    mockRemove = jest.fn();
    wrapper = shallow(
      <PomContainer pomHistory={mockPomHistory} removePom={mockRemove} />
    );
  });

  it('should matchsnapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should call removePom on click', () => {
    wrapper = mount(
      <PomContainer pomHistory={mockPomHistory} removePom={mockRemove} />
    );
    wrapper.find(i).simulate('clickl');
    expect(mockRemove).toHaveBeenCalled();
  });
});
