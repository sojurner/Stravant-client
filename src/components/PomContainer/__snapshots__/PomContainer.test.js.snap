// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PomContainer should matchsnapshot 1`] = `
<div
  className="pom-container"
>
  <div
    className="pom-cards"
  >
    <i
      class="fab fa-xbox"
      onClick={[Function]}
    />
    <h4>
      0
       
    </h4>
    <p>
       undefinedh undefinedm undefineds
    </p>
  </div>
</div>
`;
